utilities/workers/worker_manager.py

```diff
@@ -195,3 +218,35 @@ def join_workers(self) -> None:
         """
         for worker in self.__workers:
             worker.join()
+
+    def check_and_restart_dead_workers(self) -> None:
+        """
+        Check and restart dead workers.
+        """
+        for worker in self.__workers:
+            if not worker.is_alive():
+                # Log the error
+                frame = inspect.currentframe()
+                worker_name_string = self.__worker_properties.get_worker_name() + " worker"
+                self.__local_logger.error("Worker died, restarting " + worker_name_string, frame)
+
+                # Terminate and join worker
+                worker.terminate()
+                worker.join()
+                del worker
+                worker = None
```

Comment:
This seems sketchy, considering that failing to create a worker below leaves this in the list. I think it's better to copy the alive workers into a new list, and then create and append whatever number of workers is required to bring it back to the original size.